client/verta/verta/client.py

```
@@ -1536,32 +1526,26 @@ def top_k(self, key, k, ret_all_info=False, _proj_id=None, _expt_id=None):
-            raise ValueError("cannot specify both `_proj_id` and `_expt_id`")
-        elif _proj_id is None and _expt_id is None:
-            if self.__len__() == 0:
-                return self.__class__(self._conn, self._conf)
-            else:
-                expt_run_ids = self._ids
-        else:
-            expt_run_ids = None
 
-        Message = _ExperimentRunService.TopExperimentRunsSelector
-        msg = Message(project_id=_proj_id, experiment_id=_expt_id, experiment_run_ids=expt_run_ids,
-                      sort_key=key, ascending=False, top_k=k, ids_only=not ret_all_info)
-        data = _utils.proto_to_json(msg)
-        response = _utils.make_request("GET",
-                                       "{}://{}/api/v1/modeldb/experiment-run/getTopExperimentRuns".format(self._conn.scheme, self._conn.socket),
-                                       self._conn, params=data)
-        _utils.raise_for_http_error(response)
+        # apply sort to new Runs
+        new_runs = copy.deepcopy(self)
+        new_runs._find_msg.sort_key = key
+        new_runs._find_msg.ascending = False
 
-        response_msg = _utils.json_to_proto(response.json(), Message.Response)
-        if ret_all_info:
-            return response_msg.experiment_runs
-        else:
-            return self.__class__(self._conn, self._conf, [expt_run.id for expt_run in response_msg.experiment_runs])
+        # copy msg to avoid mutating `new_runs`'s state
+        msg = _ExperimentRunService.FindExperimentRuns()
+        msg.CopyFrom(new_runs._find_msg)
+        msg.page_limit = k
+        msg.page_number = 1
 
-    def bottom_k(self, key, k, ret_all_info=False, _proj_id=None, _expt_id=None):
+        response_msg = self._call_back_end(msg)
+
+        del new_runs._find_msg.experiment_run_ids[:]
+        new_runs._find_msg.experiment_run_ids.extend(record.id for record in response_msg.experiment_runs)
+
+        return new_runs
+
+    def bottom_k(self, key, k, ret_all_info=False):
```

Should we raise an error if `k` is too big? That would avoid the URL issue we had.

---

The URL issue has actually already been resolved in this PR, because it used to be a `GET` `getTopExperimentRuns`, but now it's a `POST` `findExperimentRuns` which won't pack the URI with params.